go-controller/pkg/config/config_test.go

```diff
@@ -1162,6 +1162,30 @@ mode=shared
 		gomega.Expect(err).NotTo(gomega.HaveOccurred())
 	})
 
+	It("ignores unknow fields in config file and does not return an error", func() {
+		err := ioutil.WriteFile(cfgFile.Name(), []byte(`[default]
+key=value
+
+[foobar]
+foo=bar
+`), 0644)
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+
+		app.Action = func(ctx *cli.Context) error {
+			var cfgPath string
+			cfgPath, err = InitConfig(ctx, kexec.New(), nil)
+			gomega.Expect(err).NotTo(gomega.HaveOccurred())
+			gomega.Expect(cfgPath).To(gomega.Equal(cfgFile.Name()))
```

Comment:
We seem to only be checking the path here? Can we also check what the end result for the parsed config looks like to an expected value where the foobar is ignored?

Reply:
Added one more expectation for a valid value. The only thing is, I can't check if `foo` doesn't exist, because it's not a field in the config structure.